dotcom-rendering/src/web/components/Liveness.importable.tsx

```diff
@@ -195,23 +196,26 @@ export const Liveness = ({
 	useEffect(() => {
 		if (!topOfBlog) return () => {};
 
-		const observer = new window.IntersectionObserver(
-			([entry]) => {
-				setTopOfBlogVisible(entry.isIntersecting);
+		const observer = new window.IntersectionObserver(([entry]) => {
+			const belowTopOfBlog = entry.boundingClientRect.top < 0;
 
-				if (entry.isIntersecting && onFirstPage) {
+			if (entry.isIntersecting) {
+				setTopOfBlogVisible(true);
+				setStickToast(false);
+				// If on first page, reveal blocks
+				if (onFirstPage) {
 					revealPendingBlocks();
 					setNumHiddenBlocks(0);
 					setShowToast(false);
 				}
-			},
-			{
-				root: null,
-				// A margin makes more sense because the element we're
-				// observing (topOfBlog) has no height
-				rootMargin: '100px',
-			},
-		);
+			} else {
+				setTopOfBlogVisible(false);
+				// If we're scrolling down make the toast sticky
+				if (belowTopOfBlog) {
+					setStickToast(true);
```

Comment:
I'm a bit naive on react, but would React not re-render the component every time setStickToast changes? Or is it intelligent enough to realize the value itself hasn't changed?

Reply:
> would React not re-render the component every time setStickToast changes?

This is absolutely what it does. Or at least it will re-render when state _changes_ and it will only rerender the html that needs updating. So in this case, when `stickToast` is changed it updates the css for the element, sticking or unsticking it